src/clean_data.py

```diff
@@ -0,0 +1,104 @@
+# clean_data.py
+# author: Weilin Han
+# date: 2024-10-15
+
+import pandas as pd
+import re
+import os
+
+
+def extract_column_name(raw_name_file):
+    """Extract and clean column names from .names file."""
+    
+    # Test 1: Ensure the raw name file exists, if not raise error
+    if not os.path.exists(raw_name_file):
+        raise FileNotFoundError(f"The raw_name file does not exist.")
+    
+    # Test 2: Ensure the raw name file is a .names file, if not raise error
+    if not raw_name_file.endswith('.names'):
+        raise ValueError("The raw_name file must be a .names file.")
```

Comment:
I don't think we need to be so strict here. Any file name could be fine, it's more important that it exists and that content that matters.